components/ImageSlider/ImageSlider.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { useState, useRef } from 'react';
+import {
+  Dialog, DialogPanel, DialogTitle, Transition, TransitionChild,
+} from '@headlessui/react';
+import Slider from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import { useOnClickOutside } from 'usehooks-ts';
+import { Container } from '@/components/Container';
+import { FlexBox } from '@/components/FlexBox';
+import { HeroIcon } from '@/components/HeroIcon';
+import { NextPrevButton } from '@/components/ImageSlider/NextPrevButton';
+import { Slide } from '@/components/ImageSlider/Slide';
+import { RichText } from '@/components/RichText';
+import { SrOnlyText, Text } from '@/components/Typography';
+import { type MarginType } from '@/utilities/datasource';
+import { getProcessedImage } from '@/utilities/getProcessedImage';
+import { getIsSbImagePortrait } from '@/utilities/getIsSbImagePortrait';
+import { SbSliderImageType } from '@/components/Storyblok/Storyblok.types';
+import * as styles from './ImageSlider.styles';
+
+type ImageSliderProps = React.HTMLAttributes<HTMLDivElement> & {
+  slides: SbSliderImageType[];
+  isLightText?: boolean;
+  ariaLabel?: string;
+  showExpandLink?: boolean;
+  marginTop?: MarginType;
+  marginBottom?: MarginType;
+}
+
+export const ImageSlider = ({
+  slides,
+  isLightText,
+  ariaLabel,
+  showExpandLink,
+  marginTop,
+  marginBottom,
+  ...props
+}: ImageSliderProps) => {
+  const [activeSlide, setActiveSlide] = useState(0);
+  const [pagerOffset, setPagerOffset] = useState(0);
+  const pagerWindowRef = useRef<HTMLDivElement>(null);
+  const pagerRef = useRef<HTMLUListElement>(null);
+  const sliderRef = useRef<Slider>(null);
+
+  // Modal states and refs
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const modalSliderRef = useRef<Slider>(null);
+  const modalContentRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(modalContentRef, () => {
+    setIsModalOpen(false);
+  });
+
+  const clickPrev = () => {
+    sliderRef.current?.slickPrev();
+  };
+
+  const clickNext = () => {
+    sliderRef.current?.slickNext();
+  };
+
+  const openModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+    sliderRef.current?.slickGoTo(activeSlide, true);
+  };
+
+  // This moves the thumbnail into view when the active slide changes.
+  const adjustPagerPosition = () => {
```

Comment:
It is possible to tab navigate to a pager thumbnail that is out of view and have it not be in view. Perhaps this needs to be activated on tab press and on the focused item?

Reply:
Yeah that's a good idea. I was thinking about adding a skip link to skip pass the list of thumbnails too (in the case when there are 20 images we probably don't want someone to have to tab through them all). I'll update this.